vllm/spec_decode/spec_decode_worker.py

```diff
@@ -653,6 +659,9 @@ def _run_non_driver_rank(self) -> bool:
 
         if not data["no_spec"]:
             self.scorer_worker.execute_model()
+            data = broadcast_tensor_dict(src=self._driver_rank)
```

Comment:
Extra broadcast is not ideal. But since this is a bugfix and should be low impact perf wise may have to live with it.

Reply:
Hi as discussed offline I will try to run a benchmark with and without this change to try and measure the impact of this if any.

Reply:
I tried it with Llama 405B + 8B spec model @ 32k sequence length w/ speculative_tokens = 8 and TP 8 on H100.

The results below are for `"prompt_tokens":30,"total_tokens":1054,"completion_tokens":1024` averaged over 3 runs.
Before Change: `9.321 s`
After Change:    `9.268 s`

Speedup: 99.4%

Reply:
cc: @njhill - Can we consider this acceptable as it is a necessary bugfix?

Reply:
Thanks for running the comparison. Following up on our conversation yesterday I was wondering if we can piggyback on the existing broadcast that we do. For the cp + sd case we need to broadcast the additional information that there are prefills in the speculative batch and if so run the proposer worker. For that I was wondering if we can check for prompts in the input batch and based on that we set the 'run_spec_proposer' in the initial broadcast. Looking through the proposer code I think its doing something similar when deciding which sequences to use for decoding vs prefill (https://sourcegraph.com/github.com/vllm-project/vllm/-/blob/vllm/spec_decode/top1_proposer.py?L131) .

@NickLucche we can do the is_prompt check and use that to set the  'run_spec_proposer' field in the initial broadcast itself?

Reply:
I don't know if it's sufficient to check for only prompts specifically. It doesn't go down that path when we try a single request scenario.

Reply:
Thanks for taking. Can you please share some more details on what the issue is for the single request case?

Reply:
Well, we only needed to add this extra broadcast after trying multiple requests. When we were simply sending single requests, this broadcast never came into play at all. So the condition is not simply "all prompts".